### PR TITLE
New tool entry point for scaffolding with a shorter name ('dotnet scaffold')

### PR DESCRIPTION
**Pull request by vijayrkn**
Adding @deepchoudhery  & @sayedihashimi 

@sayedihashimi - This new entry point will support everything that the current code generator supports and provides the option for adding new scaffolders. This makes use of the System.Commandline. 

*Note: Added the options only for Area to see if this is the direction we want to take. If so, we can add the options for other scaffolders as well.

### CODE REVIEW

**Comment by sayedihashimi** [comment]
Sounds good, before we release we will need to get the tool reviewed for consistency. I also have some other ideas that we can discuss.

**Comment by vijayrkn** [comment]
> Sounds good, before we release we will need to get the tool reviewed for consistency. I also have some other ideas that we can discuss.

I will set some time this week to chat more. Thanks!

**Comment by vijayrkn** [comment]
@deepchoudhery - Since this only invoked as a command line tool, we can use 'dotnet add package' to add the codegeneration package automatically to make this more seamless.

**Comment by deepchoudhery** [comment]
LGTM. Will move out the CLI descriptions either to a descriptions.json file or to project resources in the next update.

File: src/dotnet-scaffold/Program.cs [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+                aliases: new[] { "-b", "--build-base-path" },
+                description: "The build base path.")
+            {
+                IsRequired = false
+            };
+
+        private static Option NoBuildOption() =>
+            new Option<bool>(
+                aliases: new[] { "--no-build" },
+                description: "Doesn't build the project before running. It also implicitly sets the --no-restore flag.")
```
[the remark this answers is not in the document]
Comment (deepchoudhery) [reply src/dotnet-scaffold/Program.cs:125]: just for referencing

**Review by deepchoudhery — APPROVED** [review]
LGTM

**Pull request merged**